crates/turbopack-dev/src/chunking_context.rs

```diff
@@ -453,54 +456,47 @@ where
 {
     let chunks: Vec<_> = GraphTraversal::<SkipDuplicates<ReverseTopological<_>, _>>::visit(
         entries,
-        get_chunk_children,
+        |chunk: ChunkVc| async move {
+            Ok(chunk
+                .parallel_chunks()
+                .await?
+                .iter()
+                .copied()
+                .collect::<Vec<_>>()
+                .into_iter())
+        },
     )
     .await
     .completed()?
     .into_inner()
     .into_iter()
     .collect();
 
-    let chunks = ChunksVc::cell(chunks);
-    let chunks = optimize(chunks);
+    let mut ecmascript_chunks = vec![];
+    let mut css_chunks = vec![];
+    let mut other_chunks = vec![];
 
-    Ok(chunks)
-}
+    for chunk in chunks.iter() {
+        if let Some(ecmascript_chunk) = EcmascriptChunkVc::resolve_from(chunk).await? {
+            ecmascript_chunks.push(ecmascript_chunk);
+        } else if let Some(css_chunk) = CssChunkVc::resolve_from(chunk).await? {
+            css_chunks.push(css_chunk);
+        } else {
+            other_chunks.push(*chunk);
+        }
+    }
 
-/// Computes the list of all chunk children of a given chunk.
-async fn get_chunk_children(parent: ChunkVc) -> Result<impl Iterator<Item = ChunkVc> + Send> {
-    Ok(parent
-        .references()
-        .await?
+    let ecmascript_chunks =
+        optimize_ecmascript_chunks(EcmascriptChunksVc::cell(ecmascript_chunks)).await?;
+    let css_chunks = optimize_css_chunks(CssChunksVc::cell(css_chunks)).await?;
+
+    let chunks = ecmascript_chunks
```

Comment:
This creates an explicit order between JS and CSS chunks in the output asset list, which causes the changes in snapshots.